src/test/java/org/springframework/integration/kafka/dsl/KafkaDslTests.java

```diff
@@ -400,8 +401,17 @@ public IntegrationFlow channels(KafkaTemplate<Integer, String> template,
 				KafkaMessageSource<?, ?> channelSource) {
 
 			return IntegrationFlows.from(topic6Channel(template, containerFactory))
-					.channel(Kafka.publishSubscribeChannel(template, containerFactory, TEST_TOPIC7))
-					.channel(Kafka.pollableChannel(template, channelSource).id("topic8Channel"))
+					.publishSubscribeChannel(pubSub(template, containerFactory), channel -> channel
+							.subscribe(f -> f.channel(
+									Kafka.pollableChannel(template, channelSource).id("topic8Channel"))))
+					.get();
+		}
+
+		@Bean
+		public BroadcastCapableChannel pubSub(KafkaTemplate<Integer, String> template,
+				ConcurrentKafkaListenerContainerFactory<Integer, String> containerFactory) {
+
+			return (BroadcastCapableChannel) Kafka.publishSubscribeChannel(template, containerFactory, TEST_TOPIC7)
```

Comment:
Since it is still something new for this project, I would suggest to take a look into a proper generic resolution instead of casting.
I mean that `AbstractKafkaChannelSpec` should have one more generic type like `C extends AbstractKafkaChannel` and all its impls must specify a particular target channel impl. This way you probably need a new `Spec` for that `PublishSubscribeKafkaChannel` since only that one has a `BroadcastCapableChannel` impl...

Reply:
Ah - good point.

Reply:
But I think we still need the subclass because we don't want to allow `.publishSubscribeChannel(somPtpChannel, channel -> subscribe(...`

Reply:
We don't. See that spec logic:
```
protected BroadcastPublishSubscribeSpec(BroadcastCapableChannel channel) {
		Assert.state(channel.isBroadcast(),
				() -> "the " + channel +
						" must be in the 'broadcast' state for using from this 'BroadcastPublishSubscribeSpec'");
		this.target = channel;
	}

```

Reply:
Right, I saw that, but I think it's better to have the compiler disallow it.

Reply:
Yeah... OK. I see your point. If that looks easier from this project perspective, let's go ahead with subclass!